cas-server-support-saml-idp/src/main/java/org/jasig/cas/support/saml/services/idp/metadata/cache/SamlRegisteredServiceCachingMetadataResolver.java

```diff
@@ -0,0 +1,21 @@
+package org.jasig.cas.support.saml.services.idp.metadata.cache;
+
+import org.jasig.cas.support.saml.services.SamlRegisteredService;
+import org.opensaml.saml.metadata.resolver.ChainingMetadataResolver;
+
+/**
+ * This is {@link SamlRegisteredServiceCachingMetadataResolver}.
```

Comment:
Yes. This is where the SP metadata for a given service is resolved from the config and then cached via guava for a period of time.